p03-form-validations/src/Form/Form.component.js

```diff
@@ -1,20 +1,75 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
 import { Card } from '../Shared/components/Card'
 import { Button, Input } from '../Shared/components'
-import { fieldsList } from './Form.services'
+import { fieldsList, isFieldValid } from './Form.services'
 
 import "./Form.styles.css"
 
+const initialState = fieldsList.map(field=> ({...field, ready:false, value:""}))
+
 const Form = () => {
-  
+
+  const [fieldsReady, setFieldsReady] = useState(false)
+  const [listo, setListo] = useState('Not ready to send')
+  const [fieldsForm, setFieldsForm] = useState(initialState)
+
+  useEffect(()=> {
+    setFieldsReady(true)
+    setListo('Ready to send')
+    fieldsForm.map((field) =>{
```

Comment:
you can use the array.some function in order to check if at least one of the fields is not ready.

```
// search for one that has a falsy value on ready property
setFieldsReady(
  // this function returns a boolean
  fieldsForm.some(field => !field.ready)
)  
```